### PR TITLE
Only modify item.starttime when attached equals false

### PR DESCRIPTION
**Pull request by DanFerrer**
### This PR will...

- Move the logic for setting `starttime` on an item outside of `mediaController.detach()` and set it after we call the detach function 

### Why is this Pull Request needed?

- ```detatch()``` should not be responsible for setting the starttime in this case
- Prevents starttime on the new active item from being set on the item being detached in a playlist

#### Addresses Issue(s):

JW8-1688

### CODE REVIEW

**Comment by johnBartos** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  🛠 There are modified src files, but no test changes. Add tests if you're able to.
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  🗿 Set a milestone. It should be the ticket's fix version in JIRA.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/js/program/media-controller.js [hunk 1]
```diff
@@ -101,17 +101,10 @@ export default class MediaController extends Eventable {
...
     detach() {
-        const { item, mediaModel, provider } = this;
+        const { provider } = this;
         this.thenPlayPromise.cancel();
         provider.detachMedia();
         this.attached = false;
-
-        // If detaching to play a midroll (pos > 0), ensure that the player resumes at the detached time by setting starttime
-        // We don't need to do this for prerolls because the player re-attaches at time 0 by default
-        const pos = mediaModel.get('position');
```
Comment (karimJWP) [inline src/js/program/media-controller.js]: detach gets called in [`set background`](https://github.com/jwplayer/jwplayer/blob/72b1b9ced8ecf6124e37ed2a55097cff7cb208df/src/js/program/media-controller.js#L252). Is it ok if `item.starttime = pos;` is not being called when an item is being backgrounded ?

Comment (robwalch) [reply src/js/program/media-controller.js]: Yes. because backgrounded items are paused and do not need `starttime` updated to resume at the correct position.
